sklearn/model_selection/_search.py

```diff
@@ -999,3 +1004,356 @@ def fit(self, X, y=None, labels=None):
+        Parameter setting that gave the best results on the hold out data.
+
+    Notes
+    -----
+    The parameters selected are those that maximize the score of the held-out
+    data, according to the scoring parameter.
+
+    See Also
+    --------
+    :class:`GridSearchCV`:
+        Does exhaustive search over a grid of parameters.
+
+    :class:`RandomizedSearchCV`:
+        Randomized search on hyper parameters.
+
+    Examples
+    -------
+    >>> from sklearn import linear_model, gp_search, datasets
+    >>> from scipy import stats
+    >>> X, y = datasets.make_regression()
+    >>> clf = linear_model.Ridge()
+    >>> parameters = {"alpha": stats.expon(scale=100)}
+    >>> search = gp_search.GPSearchCV(clf, parameters)
+    >>> search.fit(X, y)
+    ...                             # doctest: +NORMALIZE_WHITESPACE +ELLIPSIS
+    GPSearchCV(acquisition_function='UCB', cv=None, error_score='raise',
+          estimator=Ridge(alpha=1.0, copy_X=True, fit_intercept=True, ...
+       normalize=False, random_state=None, solver='auto', tol=0.001),
+          fit_params={}, gp_params={}, iid=True, kappa=1.96, n_candidates=500,
+          n_init=3, n_iter=10, n_jobs=1,
+          param_distributions={'alpha': ...},
+          pre_dispatch='2*n_jobs', random_state=None, refit=True, scoring=None,
+          verbose=0, xi=0.01)
+
+    """
+
+    def __init__(self, estimator, param_distributions, n_init=3, n_iter=10,
+                 scoring=None, fit_params=None, acquisition_function='UCB',
+                 n_jobs=1, iid=True, refit=True,
+                 cv=None, verbose=0, pre_dispatch='2*n_jobs',
+                 random_state=None, error_score='raise', n_candidates=500,
+                 gp_params={}, xi=0.01, kappa=1.96):
+
+        if acquisition_function not in ('UCB', 'EI', 'greedy'):
+            raise ValueError(
+                "acquisition_function should be one of " +
+                "'UCB', 'EI' or 'greedy'. Found %s instead" %
+                acquisition_function)
+        self.acquisition_function = acquisition_function
+        self.param_distributions = param_distributions
+        if n_init > n_iter:
+            raise ValueError('n_init cannot be larger than n_iter')
+        self.n_init = n_init
+        self.n_iter = n_iter
+        self.random_state = random_state
+        self.gp_params = gp_params
+        if xi <= 0:
+            raise ValueError('Parameter xi must be positive')
+        self.xi = xi
+        if kappa <= 0:
+            raise ValueError('Parameter kappa must be positive')
+        self.kappa = kappa
+        self.n_candidates = n_candidates
+
+        super(SequentialSearchCV, self).__init__(
+            estimator=estimator, scoring=scoring, fit_params=fit_params,
+            n_jobs=n_jobs, iid=iid, refit=refit, cv=cv, verbose=verbose,
+            pre_dispatch=pre_dispatch, error_score=error_score)
+
+    def _evaluate_params(self, params, cv, X, y, labels):
+        all_scores = Parallel(
```

Comment:
In recent versions of joblib it may be beneficial to reuse a single `Parallel` instance rather than constructing it repeatedly.